test/build.sh

```diff
@@ -0,0 +1,146 @@
+# Set compilers to be tested depending on machine
+#-----------------------------------------------------------------------
+if [ "${machine}" == "cheyenne" ] ; then
+  compilers=( intel gnu )
+else
+  compilers=( intel )
+fi
+
+#-----------------------------------------------------------------------
+# Set some directories
+#-----------------------------------------------------------------------
+PID=$$
+TEST_DIR=$( pwd )                   # Directory with this script
+TOP_DIR=${TEST_DIR}/..              # Top level (umbrella repo) directory
+TEST_OUTPUT=${TEST_DIR}/build_test${PID}.out
+
+build_it=0        # Set to 1 to skip build (for testing pass/fail criteria)
+#-----------------------------------------------------------------------
+# Create the output file if it doesn't exist
+#-----------------------------------------------------------------------
+if [ ! -f "$TEST_OUTPUT" ]; then
+   touch ${TEST_OUTPUT}
+fi
+
+cd ${TOP_DIR}
+
+ENV_DIR=${TOP_DIR}/env
+#-----------------------------------------------------------------------
+# Arrray of all executables built
+#-----------------------------------------------------------------------
+declare -a executables_created=( chgres_cube \
+                                 emcsfc_ice_blend \
+                                 emcsfc_snow2mdl \
+                                 filter_topo \
+                                 fregrid \
+                                 fvcom_to_FV3 \
+                                 global_cycle \
+                                 global_equiv_resol \
+                                 make_hgrid \
+                                 make_solo_mosaic \
+                                 ncep_post \
+                                 orog \
+                                 orog_gsl \
+                                 regional_esg_grid \
+                                 sfc_climo_gen \
+                                 shave \
+                                 ufs_model \
+                                 vcoord_gen )
+
+#-----------------------------------------------------------------------
+# Set up the build environment and run the build script.
+#-----------------------------------------------------------------------
+  for compiler in "${compilers[@]}"; do
+    BUILD_DIR=${TOP_DIR}/build_${compiler}
+    BUILD_OUTPUT=${BUILD_DIR}/build.out
+    BIN_DIR=${TOP_DIR}/bin_${compiler}
+    EXEC_DIR=${BIN_DIR}/bin
+    if [ $build_it -eq 0 ] ; then
+      . ${ENV_DIR}/build_${machine}_${compiler}.env
+#-----------------------------------------------------------------------
+# Remove old and bin directories if they exist
+#-----------------------------------------------------------------------
+      if [ -d "${BUILD_DIR}" ] ; then rm -rf ${BUILD_DIR} ; fi
+      if [ -d "${BIN_DIR}" ] ; then rm -rf ${BIN_DIR} ; fi
+      mkdir ${BUILD_DIR}
+      cd ${BUILD_DIR}
+      cmake .. -DCMAKE_INSTALL_PREFIX=../bin_${compiler}
```

Comment:
Can this line use `${BIN_DIR}` to reduce duplication of information?

Reply:
Yes!